### ISSUE TITLE
[Question] Button and input[type="submit"] support

### ISSUE THREAD

**Issue by Jokinen** [City-of-Helsinki/helsinki-design-system#134]
**Is your feature request related to a problem? Please describe.**
I am theming a keycloak theme which comes with a set of ready made templates. At times I need to overwrite these themes to accomplish changes in content--but I try to avoid it as much as I can. Changing the template increases the complexity of version upgrades, because they will require more "merging".

In my case, these templates include forms which use `<input type="submit" />` as the control for submitting the form. The current button component in this library consists of two elements: the button and a label. Naturally I wouldn't be able to add a child node for a submit input.

**Describe the solution you'd like**
When I apply button classes to a `input[type="submit"]` I would get the same button I can find from Abstract.

**Describe alternatives you've considered**
For now, I've applied the button and label classes to the input like so

`<input type="submit" class="hds-button hds-button--primary hds-button__label" />`

and it looks to be working well enough. I am able to inject these classes through a configuration file which means that I don't need to overwrite the template itself, which is in line with my goals.

Is this a good approach? I suspect the child node is a child node for a purpose--maybe it has something to do with icons? Is there something I should look out for when using this approach?

**Comment by niglu1** [comment]
It can be done with some limitations. The focus outline is implemented using a `:after` pseudo-element, so it won't work with an `input` element. Icons wont be supported either, but maybe that isn't a problem?

So with
```html
<input type="submit" class="hds-button hds-button--primary" />
```
you would get the following (normal buttons on the left, inputs on the right)

<img width="287" alt="Screenshot 2020-05-14 at 9 19 37" src="https://user-images.githubusercontent.com/12857859/81899678-0ea45200-95c4-11ea-8f3e-cde93e0cc2b6.png">

All other classes like `--small`, `--secondary`, `--supplementary`, `--theme-black` etc. would also be supported.

I'll create a JIRA ticket for this as I kinda already implemented this.

**Comment by Jokinen** [comment]
Oh good point about the focus! I'll add some kind of temporary fix for now.

**Comment by niglu1** [comment]
Support is implemented in #136
